robot/web/web.go

```diff
@@ -626,13 +630,15 @@ func (svc *webService) startStream(streamFunc func(opts *webstream.BackoffTuning
 func (svc *webService) startImageStream(ctx context.Context, source gostream.VideoSource, stream gostream.Stream) {
 	ctxWithJPEGHint := gostream.WithMIMETypeHint(ctx, rutils.WithLazyMIMEType(rutils.MimeTypeJPEG))
 	svc.startStream(func(opts *webstream.BackoffTuningOptions) error {
-		return webstream.StreamVideoSource(ctxWithJPEGHint, source, stream, opts)
+		cancelCtx, _ := utils.MergeContext(svc.cancelCtx, ctxWithJPEGHint)
```

Comment:
ah so we still need to collect those cancels at the end of the webservice but you don't need to wait on anything now so you can cancel towards the end of the close func

Reply:
Done